lib/ansible/modules/network/onyx/onyx_ntp_servers_peers.py

```diff
@@ -0,0 +1,275 @@
+      ntp server 2.2.2.2 version 3
+      ntp server 2.2.2.2 trusted-enable
+      no ntp server 2.2.2.2
+      ntp server 192.168.10.10
+      ntpdate 192.168.10.10
+"""
+
+from copy import deepcopy
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six import iteritems
+from ansible.module_utils.network.common.utils import remove_default_spec
+
+from ansible.module_utils.network.onyx.onyx import BaseOnyxModule
+from ansible.module_utils.network.onyx.onyx import show_cmd
+
+
+class OnyxNTPServersPeersModule(BaseOnyxModule):
+
+    def init_module(self):
+        """ module initialization
+        """
+        peer_spec = dict(ip_or_name=dict(required=True),
+                         state_enabled=dict(choices=['yes', 'no']),
+                         version=dict(type='int', choices=[3, 4]),
+                         key_id=dict(type='int'),
+                         delete=dict(choices=['yes', 'no']))
+        server_spec = dict(ip_or_name=dict(required=True),
+                           state_enabled=dict(choices=['yes', 'no']),
+                           version=dict(type='int', choices=[3, 4]),
+                           trusted_enable=dict(choices=['yes', 'no']),
+                           key_id=dict(type='int'),
+                           delete=dict(choices=['yes', 'no']))
+        element_spec = dict(peers=dict(type='list', elements='dict', options=peer_spec),
+                            servers=dict(type='list', elements='dict', options=server_spec),
+                            ntpdate=dict())
+        argument_spec = dict()
+        argument_spec.update(element_spec)
+        self._module = AnsibleModule(
+            argument_spec=argument_spec,
+            supports_check_mode=True)
+
+    def get_required_config(self):
+        module_params = self._module.params
+        self._required_config = dict(module_params)
+        self.validate_param_values(self._required_config)
+
+    def _show_peers_servers_config(self):
+        cmd = "show ntp configured"
+        return show_cmd(self._module, cmd, json_fmt=True, fail_on_error=False)
+
+    def _set_servers_config(self, peers_servers_config):
+        servers = dict()
+        peers = dict()
+        if not peers_servers_config:
+            return
+        index = 0
+        for peer_server in peers_servers_config:
+            if (index == 0):
+                index += 1
+                continue
+            else:
+                header_list = peer_server.get("header").split(" ")
+                type = header_list[1]
+                if (type == 'server'):
+                    server_entry = {"version": peer_server.get("NTP version"),
+                                    "state_enabled": peer_server.get("Enabled"),
+                                    "trusted_enable": peer_server.get("Trusted"),
+                                    "key_id": peer_server.get("Key ID")}
+                    servers.update({header_list[2]: server_entry})
```

Comment:
instead of:
```python
servers.update({header_list[2]: server_entry})
```
it is more efficient to call:
```python
servers[header_list[2]] = server_entry
```